shiny/ui/_sidebar.py

```diff
@@ -312,62 +349,76 @@ def _as_open(
         if open is None:
             return None
 
-        if isinstance(open, str):
-            return SidebarOpen._from_string(open)
-
-        if isinstance(open, dict):
-            return SidebarOpen(**open)
-
         if isinstance(open, SidebarOpen):
             return open
 
-        raise ValueError("`open` must be a string or a `SidebarOpen` object")
+        return SidebarOpen._as_open(open)
 
     def _set_default_open(
         self, open: SidebarOpen | SidebarOpenSpec | SidebarOpenValue
     ) -> Sidebar:
+        """
+        Sets the default or fallback value of `open` for the sidebar. In other words,
+        this determines the fallback value of `open` if the user did not set `open` when
+        they called `sidebar()`.
+
+        This method avoids modifying the original object in place and returns a new copy
+        of the sidebar object if needed.
+        """
         new_default = self._as_open(open)
 
         if new_default is None:
             raise ValueError(
                 "`open` must be a string, a `SidebarOpen` object, or a dictionary with keys `desktop` and `mobile`."
             )
 
+        if new_default == self._default_open:
+            return self
+
```

Comment:
One more quick thought about this: almost all other Shiny components are mutable. If you have two references to a `div` and modify one of them, then the change will be reflected from the other reference as well, and the same is true for almost all other Shiny components.

Given that the name of this method, it might even be a bit of a surprise for someone that, if you do something like this:

```py
s = sidebar()
s._set_default_open("always")
```

... the second line is effectively a no-op (because the returned value wasn't stored anywhere). With a name like `._set_x`, it's very reasonable to assume that it mutates the object, instead of returning a modified copy.

Reply:
This method hides the copying to make things easier, but I think we should just have the page functions copy the object. That would add a small amount of (reasonably instructive) repetitive code and would remove the necessity for the maybe copying method.

Reply:
Done in 65ebb93115d7eef748bf0596c04a0cea15b1ba3e